apps/drain-safe/src/components/App.tsx

```diff
@@ -108,13 +113,28 @@ const App: React.FC = () => {
     }
   }, [balancesError]);
 
+  useEffect(() => {
+    sdk.eth.getGasPrice().then((gasPrice) => {
+      setGasPrice(new BigNumber(gasPrice));
+    });
+  }, [sdk.eth]);
+
+  const ethFiatPrice = Number(assets[0]?.fiatConversion || 0);
+
   return (
     <FormContainer onSubmit={onSubmit} onReset={onCancel}>
       <Flex>
         <Logo />
         <Title size="md">Drain Account</Title>
       </Flex>
-      <Balances assets={assets} exclude={excludedTokens} onExcludeChange={handleExcludeChange} />
+      <Balances
+        ethFiatPrice={ethFiatPrice}
+        gasPrice={gasPrice}
+        web3={web3}
```

Comment:
With the useWeb3 new hook you don't really need to pass it down right?

Reply:
Yep, I decided to keep the `const { web3 } = useWeb3();` and `setGasPrice` useState + useEffect in the App.tsx, But maybe makes sense to declare it directly in the Balances Component

Reply:
Really in CurrencyCell. In Balances you don't need i think

Reply:
but as we talked in the other thread we will need a context provider for avoid instance recreation. And we will address it in a different ticket.

Reply:
ahh right 😂 ... so let's improve it later